packages/babel-plugin-transform-es2015-modules-commonjs/src/index.js

```diff
@@ -206,6 +211,9 @@ export default function() {
       },
 
       Program: {
+        enter(path) {
+          this.exportNamesIdentifier = path.scope.generateUidIdentifier("exportNames");
```

Comment:
Why do this eagerly?

Reply:
iirc it simply was because I didn't see the point in generating it lazily, since it's only ever used if there's an `export *` statement in the code anyway (cf `needsExportNames`). I guess it could be `null` until we need it (which would also remove the need for `needsExportNames`).

Reply:
With this, it "uses" the name so no one else can generate it. Not a major issue.